Servlet tasks (#4)
Shotim/LeverX-Intro: task-2-SERVLETS/Self-check.md

1. What is the Servlet Container? List some of them.
2. What is the life-cycle of a servlet?
3. What are life-cycle methods for a servlet do you know and what is their functions?
4. What is REST?

> 5. What HTTP methods are used according to the REST?

Reviewer: I suppose this would be better: 
5. Which HTTP methods are used according to the REST?

Author: ```suggestion
5. Which HTTP methods are used according to the REST?
```

> 6. Describe what http-statuses are used in the responses and what server behaviour they mean according to the REST.

Reviewer: Suppose this would be better:
6. Describe which http-statuses are used in responses and what kind of server behavior do they mean according to the REST.

Author: ```suggestion
6. Describe which http-statuses are used in responses and what kind of server behavior do they mean according to the REST.
```

7. What is the difference between PUT and PATCH methods?
8. What is the Servlet Listener? What is the purpose of it?
9. What is the Servlet Filter? What is the purpose of it?

> 10. Describe the behaviour when a request is sent to the server?

Reviewer: Suppose this would be better:
10. Describe a server behavior when a request is sent to the server?

Author: You can tap on 'insert a suggestion' or ctrl-g
```suggestion
10. Describe a server behaviour when a request is sent to the server?
```

Author: And then even commit it)

> 11. What is the JSP? And what is the difference to HTML?

Reviewer 2: ```suggestion
11. What is the JSP? And what is the difference with HTML?
```

Author: According to British English, this option is correct. But for more standard view we can also use 'from'
[link](https://getproofed.com/writing-tips/word-choice-different-than-different-to-or-different-from/)

12. What are forward() and sendRedirect() methods? Difference?

> 13. What is the ServletContext and ServletConfig? What is the purpose of them?

Reviewer 2: ```suggestion
13. What is the ServletContext and ServletConfig? What's their purpose?
```
